checkers_bot_tournament/game_result.py

```diff
@@ -1,19 +1,88 @@
 from dataclasses import dataclass
-from typing import Optional
+from enum import auto, Enum
+import inspect
+from typing import Optional, Literal
+
+class Result(Enum):
+    WHITE = auto()
+    BLACK = auto()
+    DRAW = auto()
 
 @dataclass
 class GameResult:
     game_id: int
     game_round: int
-    winner_name: str
-    winner_colour: str
-    winner_kings_made: int
-    winner_num_captures: int
-    loser_name: str
-    loser_colour: str
-    loser_kings_made: int
-    loser_num_captures: int
+
+    result: Result
+
+    white_name: str
+    white_kings_made: int
+    white_num_captures: int
+
+    black_name: str
+    black_kings_made: int
+    black_num_captures: int
+
     num_moves: int
     # This is quite chunky (it stores every move and board state) so it's optional
     moves: Optional[str]
-    
\ No newline at end of file
+    
+    def result_summary(self) -> str:
+        if self.result == Result.DRAW:
+            # Arbitrarily White is player 1
+            player1_stats = self.white_summary("White Player Details:")
+            player2_stats = self.black_summary("Black Player Details:")
+        elif self.result == Result.WHITE:
+            player1_stats = self.white_summary("Winner Details:")
+            player2_stats = self.black_summary("Loser Details:")
+        else:
+            player1_stats = self.black_summary("Winner Details:")
+            player2_stats = self.white_summary("Loser Details:")
```

Comment:
is there a point of having numbered players instead of something like `white_player` `black_player`?

Reply:
Depends if whether you want to display winner then loser for decisive games (obvs there are no winners in a draw so white arbitrarily gets displayed first), or always one colour then another regardless of result.